cmd/cfssl/cfssl_test.go

```diff
@@ -1 +1,446 @@
+	Success bool `json:"success"`
+	Result  struct {
+		CLIScanResponse
+	} `json:"result"`
+}
+
+// API Struct for CertInfo utilizes certinfo.Certificate struct.
+type APICertInfoResponse struct {
+	Success bool `json:"success"`
+	Result  struct {
+		certinfo.Certificate
+	} `json:"result"`
+}
+
+// API Struct for Sign and Info will utilize info.Resp struct (for the
+// "certificate" field)
+type APIResponse struct {
+	Success bool `json:"success"`
+	Result  struct {
+		info.Resp
+	} `json:"result"`
+}
+
+// CFSSL Sign on CLI returns a json with a "cert" field, not "certificate"
+// as is with the Sign API response struct defined in scan_common.go.
+// The following struct is made to suit that situation.
+type CLISignResponse struct {
+	Certificate string `json:"cert"`
+}
+
+// Start up the CFSSL local server.
+func setup() {
+	// Set up a test server using our CA certificate and key.
+	CACert, CAKey, err := testsuite.CreateSelfSignedCert(CARequest)
+	if err != nil {
+		panic(err)
+	}
+	// Set up a test server using our CA certificate and key.
+	serverData := testsuite.CFSSLServerData{CA: CACert, CAKey: CAKey}
+	s, err := testsuite.StartCFSSLServer(addressToTest, portToTest,
+		serverData)
+	server = s
+	if err != nil {
+		setdown()
+		panic(err)
+	}
+}
+
+// Kill the CFSSL local server.
+func setdown() {
+	err := server.Kill()
+	if err != nil {
+		panic(err)
+	}
+}
+
+// Return API output
+func apiOut(request string, jsonStr []byte,
+	endpoint string, t *testing.T) []byte {
+	setup()
+	// Generate request
+	req, err := http.NewRequest(request, endpoint,
+		bytes.NewBuffer(jsonStr))
+	if err != nil {
+		setdown()
+		t.Fatalf("Error creating new HTTP request, %v", err)
+	}
+	client := &http.Client{}
+	// Execute request
+	resp, err := client.Do(req)
+	if err != nil {
+		setdown()
+		t.Fatalf("Error sending request to client, %v", err)
+	}
+	defer resp.Body.Close()
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		setdown()
+		t.Fatalf("Error reading API response, %v", err)
+	}
+	setdown()
+	// return API response to the request
+	return body
+}
+
+// Return CLI output
+func cliOut(command *exec.Cmd, t *testing.T) []byte {
+	CLIOutput, err := command.Output()
+	if err != nil {
+		t.Fatalf("Error sending request %s to cli, %v",
+			string(CLIOutput), err)
+	}
+	return []byte(CLIOutput)
+}
+
+// Format the API/CLI Output into the following structs
+func apiFormat(body []byte) APIResponse {
+	apiRes := APIResponse{}
+	json.Unmarshal(body, &apiRes)
+	return apiRes
+}
+
+func responseUnmarshal(body []byte) api.Response {
+	res := api.Response{}
+	json.Unmarshal(body, &res)
+	return res
+}
+
+// START OF TESTS
+
+// Testing the Bundle API against CFSSL Bundle CLI
+func TestBundle(t *testing.T) {
+	// bundleAPIResp holds response from bundle endpoint
+	jsonStr := []byte(`{"domain": "cloudflare.com"}`)
+	body := apiOut("POST", jsonStr, postStr+"bundle", t)
+	bundleAPIResp := APIBundleResponse{}
+	json.Unmarshal(body, &bundleAPIResp)
```

Comment:
If invalid JSON is returned from the API then it will be harder to debug. I think you should do `if err := json.Unmarshal(body, &bundleAPIResp); err != nil {` whenever data in unmarshalled